worlds/oot_soh/location_access/overworld/thieves_hideout.py

```diff
@@ -3,7 +3,233 @@
+                        and has_item(LocalEvents.TH_DOUBLE_CELL_COULD_FREE_DOUBLE_CELL_CARPENTER, bundle)
+                        and has_item(LocalEvents.TH_STEEP_SLOPE_CELL_COULD_FREE_SLOPE_CARPENTER, bundle)
+                        and has_item(LocalEvents.TH_DEAD_END_CELL_COULD_FREE_DEAD_END_CARPENTER, bundle)
+                        and has_item(LocalEvents.TH_1_TORCH_CELL_COULD_FREE_1_TORCH_CARPENTER, bundle))
+                        or world.options.fortress_carpenters.value == 2)
+    ])
+
+    ##Thieves Hideout Double Cell
+    # Events
+    add_events(Regions.THIEVES_HIDEOUT_DOUBLE_CELL, world, [
+        (EventsLocations.TH_DOUBLE_CELL_FREE_DOUBLE_CELL_CARPENTER, LocalEvents.TH_DOUBLE_CELL_COULD_FREE_DOUBLE_CELL_CARPENTER,
+         lambda bundle: can_kill_enemy(bundle, Enemies.GERUDO_WARRIOR)),
+    ])
+    # Locations
+    add_locations(Regions.THIEVES_HIDEOUT_DOUBLE_CELL, world, [
+        (Locations.TH_DOUBLE_CELL_CARPENTER, lambda bundle: can_kill_enemy(bundle, Enemies.GERUDO_WARRIOR)),
+        (Locations.TH_NEAR_DOUBLE_CELL_RIGHT_POT, lambda bundle: can_break_pots(bundle)),
+        (Locations.TH_NEAR_DOUBLE_CELL_MID_POT, lambda bundle: can_break_pots(bundle)),
+        (Locations.TH_NEAR_DOUBLE_CELL_LEFT_POT, lambda bundle: can_break_pots(bundle)),
+        (Locations.TH_RIGHTMOST_JAILED_POT, lambda bundle: can_break_pots(bundle)),
+        (Locations.TH_RIGHT_MIDDLE_JAILED_POT, lambda bundle: can_break_pots(bundle)),
+        (Locations.TH_LEFT_MIDDLE_JAILED_POT, lambda bundle: can_break_pots(bundle)),
+        (Locations.TH_LEFTMOST_JAILED_POT, lambda bundle: can_break_pots(bundle)),
+        (Locations.TH_DOUBLE_CELL_LEFT_CRATE, lambda bundle: can_break_crates(bundle)),
+        (Locations.TH_DOUBLE_CELL_RIGHT_CRATE, lambda bundle: can_break_crates(bundle))
+    ])
+    # Connections
+    connect_regions(Regions.THIEVES_HIDEOUT_DOUBLE_CELL, world, [
+        (Regions.GERUDO_FORTRESS_OUTSKIRTS, lambda bundle: True),
+        (Regions.GF_NEAR_GROTTO, lambda bundle: True),
+        (Regions.THIEVES_HIDEOUT_RESCUE_CARPENTERS,
+         lambda bundle: (((small_keys(Items.GERUDO_FORTRESS_SMALL_KEY, 4,
+                                      bundle) and world.options.fortress_carpenters.value == 0)
+                          or (small_keys(Items.GERUDO_FORTRESS_SMALL_KEY, 1,
+                                         bundle) and world.options.fortress_carpenters.value == 1))
+                         and has_item(LocalEvents.TH_DOUBLE_CELL_COULD_FREE_DOUBLE_CELL_CARPENTER, bundle)
+                         and has_item(LocalEvents.TH_STEEP_SLOPE_CELL_COULD_FREE_SLOPE_CARPENTER, bundle)
+                         and has_item(LocalEvents.TH_DEAD_END_CELL_COULD_FREE_DEAD_END_CARPENTER, bundle)
+                         and has_item(LocalEvents.TH_1_TORCH_CELL_COULD_FREE_1_TORCH_CARPENTER, bundle))
+                        or world.options.fortress_carpenters.value == 2)
+    ])
```

Comment:
I'm ok with both the event and the gerudo membership location being in this new region, but you should move the logic rules from all the connections to that new region to the event in that new region instead. Then you can check for having the event in the location in that new region too, so you'd only have to write this entire logic rule once.

Reply:
Its because the skulltula had different rules depending on from what region it tried to connect to the skulltula. In this case the rule is always the exact same.